packages/node_modules/pouchdb-core/src/active-tasks.js

```diff
@@ -0,0 +1,48 @@
+import {v4 as uuidv4} from "uuid";
+
+export default class ActiveTasks {
+  constructor() {
+    this.tasks = {};
+  }
+
+  list() {
+    return Object.values(this.tasks);
+  }
+
+  add(task) {
+    const id = uuidv4();
+    this.tasks[id] = {
+      id,
+      name: task.name,
+      total_items: task.total_items,
+      created_at: new Date().toJSON()
+    };
+    return id;
+  }
+
+  get(id) {
+    return this.tasks[id];
+  }
+
+  /* eslint-disable no-unused-vars */
+  remove(id, reason) {
+    delete this.tasks[id];
+    return this.tasks;
+  }
+
+  update(id, updatedTask) {
```

Comment:
This implementation is a nice protection against race conditions because it replaces the task with a _new object_. That means any code that's holding onto a task it got via `activeTasks.get()` won't mess anything up by mutating that object, and updates need to go via the `update()` method in order for other code to see them.